ConferencesApp/ConferenceTableViewController.swift

```diff
@@ -117,11 +130,107 @@ class ConferenceTableViewController: UITableViewController, MKMapViewDelegate {
 
         let theSpan:MKCoordinateSpan = MKCoordinateSpanMake(latDelta, longDelta)
         let venueLocation:CLLocationCoordinate2D = CLLocationCoordinate2DMake(latitude, longitude)
-                let theRegion:MKCoordinateRegion = MKCoordinateRegionMake(venueLocation, theSpan)
-
+        let theRegion:MKCoordinateRegion = MKCoordinateRegionMake(venueLocation, theSpan)
+        
         return theRegion
     }
-
+    
+    func setTapGestureOnDateImage() {
+        let tapGesture = UITapGestureRecognizer()
+        tapGesture.numberOfTapsRequired = 1
+        tapGesture.addTarget(self, action: "tapOnDateImageHandler:")
+        dateImage.addGestureRecognizer(tapGesture)
+    }
+    
+    func tapOnDateImageHandler(sender: UITapGestureRecognizer) {
+        let eventStore = EKEventStore()
+        
+        eventStore.requestAccessToEntityType(EKEntityType.Event, completion: {
+            granted, error in
+            if (granted) && (error == nil) {
+                if self.calendarEventID.isEmpty {
+                     dispatch_async(dispatch_get_main_queue()) {
+                        self.addCalendarEventtoStore(eventStore)
+                    }
+                    
+                } else { // Event Already present in calendar.
+                    dispatch_async(dispatch_get_main_queue()) {
+                        self.deleteEventFromStore(eventStore, eventID:self.calendarEventID)
+                    }
+                }
+            }
+        })
+        
+    }
+    
+    func addCalendarEventtoStore(eventStore:EKEventStore) {
```

Comment:
Colon Violation: When specifying a type, always associate the colon with the identifier